cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -163,99 +150,308 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             ;
     }
 
+    static void setitem_parser(wrapped_type & arr_out, pybind11::args & args)
+    {
+        namespace py = pybind11;
+
+        if (args.size() == 2)
+        {
+            // sarr[K] = V
+            if (py::isinstance<py::int_>(args[0]) &&
+                (py::isinstance<py::bool_>(args[1]) || py::isinstance<py::int_>(args[1]) ||
+                 py::isinstance<py::float_>(args[1])))
+            {
+                const auto key = args[0].cast<ssize_t>();
+
+                arr_out.at(key) = args[1].cast<T>();
+                return;
+            }
+            // sarr[K1, K2, K3] = V
+            if (py::isinstance<py::tuple>(args[0]) &&
+                (py::isinstance<py::bool_>(args[1]) || py::isinstance<py::int_>(args[1]) ||
+                 py::isinstance<py::float_>(args[1])))
+            {
+                const auto key = args[0].cast<std::vector<ssize_t>>();
+
+                arr_out.at(key) = args[1].cast<T>();
+                return;
+            }
+            // multi-dimension with slice and ellipsis
+            // sarr[slice, slice, ellipsis] = ndarr
+            if (py::isinstance<py::tuple>(args[0]) && py::isinstance<py::array>(args[1]))
+            {
+                const auto tuple = args[0].cast<py::tuple>();
```

Comment:
Maybe `const py::tuple args_in = args[0];` would just work.  It looks a little bit strange to cast a Python type to another Python type.

Reply:
And naming a variable using its type name is hard to read.